libhb/hb_dict.c

```diff
@@ -440,19 +446,234 @@ hb_dict_t * hb_dict_init()
 {
-    return json_object_get(dict, key);
+    hb_value_t * result;
+
+    // First try case sensitive lookup
+    result = json_object_get(dict, key);
+    if (result == NULL)
+    {
+        // If not found, try case insensitive lookup
+        char * lower = makelower(key);
+        result = json_object_get(dict, lower);
+        free(lower);
+    }
+    return result;
+}
+
+//  Dictionary extraction helpers
+//
+// Extract the given key from the dict and assign to dst *only*
+// if key is found in dict.  Values are converted to the requested
+// data type.
+//
+// return: 1 - key is in dict
+//         0 - key is not in dict
+int hb_dict_extract_int(int *dst, const hb_dict_t * dict, const char * key)
+{
+    if (dict == NULL || key == NULL || dst == NULL)
+    {
+        return 0;
+    }
+
+    hb_value_t *val = hb_dict_get(dict, key);
+    if (val == NULL)
+    {
+        return 0;
+    }
+
+    *dst = hb_value_get_int(val);
+    return 1;
+}
+
+int hb_dict_extract_double(double *dst, const hb_dict_t * dict,
+                                        const char * key)
+{
+    if (dict == NULL || key == NULL || dst == NULL)
+    {
+        return 0;
+    }
+
+    hb_value_t *val = hb_dict_get(dict, key);
+    if (val == NULL)
+    {
+        return 0;
+    }
+
+    *dst = hb_value_get_double(val);
+    return 1;
+}
+
+int hb_dict_extract_bool(int *dst, const hb_dict_t * dict, const char * key)
+{
+    if (dict == NULL || key == NULL || dst == NULL)
+    {
+        return 0;
+    }
+
+    hb_value_t *val = hb_dict_get(dict, key);
+    if (val == NULL)
+    {
+        return 0;
+    }
+
+    *dst = hb_value_get_bool(val);
+    return 1;
+}
+
+int hb_dict_extract_string(char **dst, const hb_dict_t * dict, const char * key)
+{
+    if (dict == NULL || key == NULL || dst == NULL)
+    {
+        return 0;
+    }
+
+    hb_value_t *val = hb_dict_get(dict, key);
+    if (val == NULL)
+    {
+        return 0;
+    }
+
+    *dst = hb_value_get_string_xform(val);
+    return 1;
+}
+
+int hb_dict_extract_rational(hb_rational_t *dst, const hb_dict_t * dict,
+                                                 const char * key)
+{
+    if (dict == NULL || key == NULL || dst == NULL)
+    {
+        return 0;
+    }
+
+    hb_value_t *val = hb_dict_get(dict, key);
+    if (val == NULL)
+    {
+        return 0;
+    }
+
+    if (hb_value_type(val) == HB_VALUE_TYPE_DICT)
+    {
+        hb_value_t * num_val = hb_dict_get(val, "Num");
+        if (num_val == NULL)
+        {
+            return 0;
+        }
+        hb_value_t * den_val = hb_dict_get(val, "Num");
+        if (den_val == NULL)
+        {
+            return 0;
+        }
+
+        dst->num = hb_value_get_int(num_val);
+        dst->den = hb_value_get_int(den_val);
+    }
+    else if (hb_value_type(val) == HB_VALUE_TYPE_STRING)
+    {
+        const char * str = hb_value_get_string(val);
+        char ** rational = hb_str_vsplit(str, '/');
```

Comment:
Good eye.  Thanks.  I did run it through valgrind, but I didn't try a framerate option that invoked this code path.